CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -1588,26 +1799,53 @@ if userGuiIcon3D and vr3dGuis then
 end
 
 
-local function MoveHamburgerTo3D()
-	Topbar3DPanel:SetType(Panel3D.Type.Fixed)
-	Topbar3DPanel:LinkTo("Backpack")
-	Topbar3DPanel:SetVisible(true)
+local function EnableVR()
+	local VRHub = require(GuiRoot.Modules.VR.VRHub)
+
+	TopbarPanel3D:SetType(Panel3D.Type.Fixed)
+	TopbarPanel3D:LinkTo("Backpack")
+	TopbarPanel3D:SetVisible(true)
 
 	local backpackPanel = Panel3D.Get("Backpack")
 	local panelLocalCF = CFrame.Angles(math.rad(5), 0, 0) * CFrame.new(0, -1.625, 0) * CFrame.Angles(math.rad(5), 0, 0)
-	function Topbar3DPanel:PreUpdate(cameraCF, cameraRenderCF, userHeadCF, lookRay)
+	function TopbarPanel3D:PreUpdate(cameraCF, cameraRenderCF, userHeadCF, lookRay)
 		local panelOriginCF = backpackPanel.localCF or CFrame.new()
 		self.localCF = panelOriginCF * panelLocalCF
 	end
 
-	function Topbar3DPanel:OnUpdate()
-		for _, item in pairs(ThreeDMenubar:GetItems()) do
+	function TopbarPanel3D:OnUpdate()
+		for _, item in pairs(Menubar3D:GetItems()) do
 			item:SetTransparency(self.transparency)
 		end
 	end
 
+	local modulesKeepingTopbarOpen = {}
+	local numModulesKeepingTopbarOpen = 0
+
+	VRHub.ModuleOpened.Event:connect(function(moduleName, isExclusive, shouldCloseNonExclusive, shouldKeepTopbarOpen)
+		if shouldKeepTopbarOpen then
+			TopbarPanel3D:SetCanFade(false)
+			TopbarPanel3D:SetVisible(true)
+
+			modulesKeepingTopbarOpen[moduleName] = true
+			numModulesKeepingTopbarOpen = numModulesKeepingTopbarOpen + 1
+		end
+	end)
+	VRHub.ModuleClosed.Event:connect(function(moduleName)
+		if modulesKeepingTopbarOpen[moduleName] then
+			modulesKeepingTopbarOpen[moduleName] = nil
+			numModulesKeepingTopbarOpen = math.max(0, numModulesKeepingTopbarOpen - 1)
```

Comment:
This feels like it could get out of sync somehow. If we could somehow ask our modules if they are open / need to keep topbar open that would feel better. Plus if we did that we could get rid of the `shouldKeepTopbarOpen` variable